modules/transport-netty4/src/test/java/org/elasticsearch/transport/SharedGroupFactoryTests.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.transport;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.http.netty4.Netty4HttpServerTransport;
+import org.elasticsearch.test.ESTestCase;
+
+public final class SharedGroupFactoryTests extends ESTestCase {
+
+    public void testSharedEventLoops() throws Exception {
+        SharedGroupFactory sharedGroupFactory = new SharedGroupFactory(Settings.EMPTY);
+
+        SharedGroupFactory.SharedGroup httpGroup = sharedGroupFactory.getHttpGroup();
+        SharedGroupFactory.SharedGroup transportGroup = sharedGroupFactory.getTransportGroup();
+
+        try {
+            assertSame(httpGroup.getLowLevelGroup(), transportGroup.getLowLevelGroup());
+        } finally {
+            httpGroup.shutdown();
+            assertFalse(httpGroup.getLowLevelGroup().isShuttingDown());
+            assertFalse(transportGroup.getLowLevelGroup().isShuttingDown());
+            transportGroup.shutdown();
+            assertTrue(httpGroup.getLowLevelGroup().isShuttingDown());
+            assertTrue(transportGroup.getLowLevelGroup().isShuttingDown());
```

Comment:
can you also check the termination future? i.e. whether the shutdown has actually completed here?